docs/custom-plugins/extended-types.md

```diff
@@ -0,0 +1,74 @@
+# Type extensions in plugins
+
+Redocly CLI in its core has a type tree which defines the structure of the API definition.
+Redocly CLI then uses it to do a type-aware traversal of an OpenAPI definition.
```

Comment:
question: Is the plan to update this from saying "OpenAPI" once this merges? https://github.com/Redocly/redocly-cli/pull/1127

Reply:
I think we'll need to update a few things, yes! I intentionally made things more vague in a few places but at this time, I think the type extension feature is really only useful for OpenAPI.